### DISCUSSION TITLE
Correct grain boundaries length calculation

### DISCUSSION THREAD

**Discussion by NikitaSakharov** [mtex-toolbox/mtex#1209] [Ask Anything]
It seems to me that grain boundaries length, which is calculated by 'sum(gB.segLength)' command is not correct.

In fact it does the same as the following code does

sum1 = 0
for i = 1 : length(gB)
sum1 = sum1 + sum(gB(i).segLength)
end

But it is not correct when I want to get the length of all boundaries.
It is not correct because each real boundary is shared by two grains. So each 'gB(i).segLength' must be divided by some number. This number depends on how many segments of gB(i) are shared by two grains. 

I'm not sure the explaination is clear, because of my english language.

But if it is clear enough, how can we calculate real grain boundaries length, not exaggerated one?

**Comment by nyyssont** [comment]
When you calculate the boundary segments for each individual grain sequentially, you will calculate each grain boundary twice. However, if you get the boundaries for the whole grain map at once (or just your selected subset of grains), you don't get duplicate boundaries.

Here is some code you can try:
```
mtexdata csl
plotx2east
[grains,ebsd.grainId] = calcGrains(ebsd('indexed'));
%%
%List of neighbors
neighs = grains.neighbors;

%Random neighbors
rand_neighs = neighs(randi(length(neighs)),:)

%Sum the boundary segments taken for each grain individually:
sum1 = sum([grains(rand_neighs(1)).boundary.segLength ;...
    grains(rand_neighs(2)).boundary.segLength])

%Sum the boundaries taken at once:
sum2 = sum(grains(rand_neighs).boundary.segLength)

%Difference:
difbound = sum1 - sum2

%total boundary
boundneighs = grains(rand_neighs).boundary;
boundshared = boundneighs(ismember(boundneighs.grainId,rand_neighs,'rows'));

%This value should equal the difference:
boundsharedlength = sum(boundshared.segLength)

%plot
plot(grains(rand_neighs))
hold on
plot(boundshared,'linecolor','red')
```

**Reply by NikitaSakharov** [reply]
Tuomo, it's clear now.

Many thanks for an explanation!

Sincerely, Nikita
